frontend/src/services/GooglePlacesService.ts

```diff
@@ -0,0 +1,183 @@
+import axios, { AxiosError, AxiosResponse } from "axios";
+import * as dotenv from "dotenv";
+import { RestaurantDetails } from ".././types/RestaurantDetails";
+
+/* eslint-disable  no-console */
+// todo: remove this eslint-disable
+class GooglePlacesService {
+  public currRestaurants: Map<string, RestaurantDetails>;
+
+  constructor() {
+    dotenv.config();
+    this.currRestaurants = new Map();
+  }
+
+  getRestaurantPlaceID(
+    restaurantName: string,
+    restaurantLocation: string
+  ): Promise<string> {
+    const dataURI: string = encodeURIComponent(restaurantName);
+    let responseData: any[];
+
+    return axios
+      .get(
+        "https://maps.googleapis.com/maps/api/place/findplacefromtext/json?",
+        {
+          params: {
+            input: dataURI,
+            inputtype: "textquery",
+            fields: "place_id",
+            key: process.env.REACT_APP_GOOGLE_PLACES_API_KEY,
+          },
+        }
+      )
+      .then((response: AxiosResponse) => {
+        if (response.data.candidates.length === 1) {
+          return Promise.resolve(response.data.candidates?.[0].place_id);
+        }
+        const promises: Array<Promise<boolean>> = [];
+        responseData = response.data.candidates;
+
+        for (const currPlace of response.data.candidates) {
+          promises.push(
+            this.isCorrectRestaurantLocation(
+              restaurantLocation,
+              currPlace.place_id
+            )
+          );
+        }
+
+        return Promise.all(promises);
+      })
+      .then((result) => {
+        if (!Array.isArray(result)) {
+          return Promise.resolve(result);
+        }
+
+        for (let i = 0; i < result.length; i++) {
+          if (result[i]) {
+            return Promise.resolve(responseData[i].place_id);
+          }
+        }
+
```

Comment:
Does google places follow some kind of format where the element in the first array is the one that matches the location + name the best? If it is, I would recommend we just choose the first one at index 0.

Reply:
If not, this will be more complicated because of comment mentioned below

Reply:
Ya, this is a good suggestion. @ld357 If this isn't possible, then I think we can go ahead with this implementation as is but I think I'll create a ticket to look into something like https://developers.google.com/places/web-service/autocomplete but we could probably discuss this more on Saturday!

Reply:
Ahhh yeah I was hoping to do that too when I was researching into it, but it isn't possible 😢   Right now I'm doing a Places Search - Find Place request (which takes in a name, address, or phone number) and then a Places Details request to check the location match. I decided to take in the name since I feel like it's less likely to be incorrect compared to an address.

There is another one called "Places API Text Search Service" which allows searches like "tacos in Vancouver" or "123 E Broadway" which will return a list of places that seem to match. I don't think it's the best idea though because it's more generic and relies heavily on the user's submitted address.